android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageView.kt

```diff
@@ -0,0 +1,424 @@
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.imePadding
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.pager.HorizontalPager
+import androidx.compose.foundation.pager.rememberPagerState
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.rxjava3.subscribeAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
+import kotlinx.coroutines.launch
+import org.uwaterloo.subletr.R
+import org.uwaterloo.subletr.api.models.ListingDetails
+import org.uwaterloo.subletr.api.models.ResidenceType
+import org.uwaterloo.subletr.components.button.PrimaryButton
+import org.uwaterloo.subletr.components.button.SecondaryButton
+import org.uwaterloo.subletr.theme.SubletrTheme
+import org.uwaterloo.subletr.theme.SubletrTypography
+import org.uwaterloo.subletr.theme.primaryTextColor
+import org.uwaterloo.subletr.theme.secondaryButtonBackgroundColor
+import org.uwaterloo.subletr.theme.textFieldBackgroundColor
+import org.uwaterloo.subletr.theme.textOnSubletrPink
+import java.time.OffsetDateTime
+import java.time.format.DateTimeFormatter
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun ListingDetailsPageView(
+	modifier: Modifier = Modifier,
+	viewModel: ListingDetailsPageViewModel = hiltViewModel(),
+	uiState: ListingDetailsPageUiState = viewModel.uiStateStream.subscribeAsState(
+		ListingDetailsPageUiState.Loading
+	).value,
+	@Suppress("UNUSED_PARAMETER")
+	listingId: String?,
```

Comment:
I think you can just remove this parameter?
You're using `SavedStateHandle`, right?

Reply:
yeah just found out it _can_ be removed